custom_image_utils/args_inferer.py

```diff
@@ -128,12 +128,22 @@ def _get_dataproc_image_path_by_version(version):
   """Get Dataproc base image name from version."""
   # version regex already checked in arg parser
   parsed_version = version.split(".")
-  filter_arg = "--filter=labels.goog-dataproc-version=\'{}-{}-{}\'".format(
-      parsed_version[0], parsed_version[1], parsed_version[2])
-  command = [
-      "gcloud", "compute", "images", "list", "--project", "cloud-dataproc",
-      filter_arg, "--format=csv[no-heading=true](name,status)"
-  ]
+  if len(parsed_version) == 2:
+    parsed_version[1] = parsed_version[1].replace('-', '-\d+-')
+    filter_arg = "--filter=labels.goog-dataproc-version ~ {}-{}".format(parsed_version[0], 
```

Comment:
I think you want to use `^{}-{}-` instead of `{}-{}` as the pattern, otherwise it might mismatch unwanted images.